src/middlewared/middlewared/plugins/iscsi_/alua.py

```diff
@@ -0,0 +1,390 @@
+import asyncio
+import itertools
+
+from middlewared.service import Service, job
+
+CHUNK_SIZE = 20
+RETRY_SECONDS = 5
+HA_TARGET_SETTLE_SECONDS = 10
+GET_UNIT_STATE_SECONDS = 2
+
+
+class iSCSITargetAluaService(Service):
+    """
+    Support iSCSI ALUA configuration.
+
+    The ALUA mechanism is based up DLM support baked into SCST, along with other
+    potions of middleware (dlm, iscsi.targets, etc ) to handle the coordination
+    between the two nodes in a HA pair.  This is performed in response to
+    cluster_mode being set on target extents.
+
+    However, when a LARGE number of extents(/targets) are present it becomes
+    impractical to leave/enter lockspaces on scst startup.
+
+    To avoid this, SCST on the ACTIVE will start without cluster_mode being
+    set on extents.  Likewise on the STANDBY node, so the targets there will be
+    present but disabled.  However, the STANDBY will then initiate a job to
+    (gradually) enable cluster_mode on the ACTIVE and react.
```

Comment:
We should probably have explicit checks within the methods to ensure that failover.status is the expected value when starting jobs that should only be on active or standby controller.

Reply:
Both `iscsi.alua.active_elected` and `iscsi.alua.activate_extents` are launched from `vrrp_master` before the `failover.status_refresh` call is made.  So, not really an option.